libs/blocks/global-navigation/utilities/keyboard/index.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable class-methods-use-this */
+import { getNextVisibleItem, getPreviousVisibleItem, selectors } from './utils.js';
+import MainNav from './mainNav.js';
+import { closeAllDropdowns } from '../utilities.js';
+
+const cycleOnOpenSearch = ({ e, isDesktop }) => {
+  const withoutBreadcrumbs = [
+    ...document.querySelectorAll(`
+  ${selectors.brand}, 
+  ${selectors.mainNavToggle},
+  ${selectors.mainNavItems},
+  ${selectors.searchTrigger},
+  ${selectors.searchField},
+  ${selectors.signIn},
+  ${selectors.profileButton},
+  ${selectors.logo}
+  `),
+  ];
+  const first = getNextVisibleItem(-1, withoutBreadcrumbs);
+  const last = getPreviousVisibleItem(withoutBreadcrumbs.length, withoutBreadcrumbs);
+  const openSearch = isDesktop && document.querySelector(selectors.openSearch);
+  if (openSearch && document.activeElement === withoutBreadcrumbs[e.shiftKey ? first : last]) {
+    e.preventDefault();
+    withoutBreadcrumbs[e.shiftKey ? last : first].focus();
+  }
+};
+
+const closeOnClickOutside = (e) => {
+  if (
+    !e.target.closest('header')
+    || e.target.classList.contains(selectors.curtain.replace('.', ''))
+  ) {
+    closeAllDropdowns();
+  }
+};
+
+class KeyboardNavigation {
+  constructor() {
+    this.addEventListeners();
+    this.mainNav = new MainNav();
+    this.desktop = window.matchMedia('(min-width: 900px)');
+  }
+
+  addEventListeners = () => {
+    document.addEventListener('click', (e) => closeOnClickOutside(e));
+
+    document.querySelector('header').addEventListener('keydown', (e) => {
+      if (e.shiftKey && e.code === 'Tab') {
+        cycleOnOpenSearch({ e, isDesktop: this.desktop.matches });
+        return;
+      }
+      switch (e.code) {
+        case 'Tab': {
+          cycleOnOpenSearch({ e, isDesktop: this.desktop.matches });
+          break;
+        }
+        case 'Enter': {
+          e.preventDefault();
+          e.target.click();
+          break;
+        }
+        case 'Escape': {
+          const profileBtn = document.querySelector(selectors.profileButton);
+          if (e.target === profileBtn && profileBtn.getAttribute('aria-expanded') === 'true') {
+            profileBtn.setAttribute('aria-expanded', 'false');
+          }
+          break;
```

Comment:
Could this be handled in the Profile block logic directly, as we currently do with the Search block escape listeners?